libraries/crypto/src/cbc.rs

```diff
@@ -13,24 +13,21 @@
 // limitations under the License.
 
 use super::util::{xor_block_16, Block16};
-use super::{Decrypt16BytesBlock, Encrypt16BytesBlock};
+use crate::aes256::{DecryptionKey, EncryptionKey};
+use core::convert::TryInto;
 
-pub fn cbc_encrypt<K>(key: &K, mut iv: Block16, blocks: &mut [Block16])
-where
-    K: Encrypt16BytesBlock,
-{
-    for block in blocks {
+pub fn cbc_encrypt(key: &EncryptionKey, mut iv: Block16, blocks: &mut [u8]) {
```

Comment:
Nit: Given that the input doesn't take a slice of blocks anymore but a slice of bytes, it might make sense to rename the `blocks` parameter as something like `bytes`, as the new API doesn't guarantee the constraint that this parameter is a multiple of 16-byte blocks.

It would also be worth documenting what the behavior should be if the input is not a multiple of 16 bytes (i.e. the "remainder" is left as is rather than encrypted, which might be surprising at best, and at worst could lead to a security issue).

Reply:
IMO it would be more robust to have an API that returns a `Result` or `panic`s if the input is not a multiple of 16 bytes.